tests/Credentials/ImpersonatedServiceAccountCredentialsTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Google\Auth\Tests\Credentials;
+
+use Google\Auth\Credentials\ImpersonatedServiceAccountCredentials;
+use LogicException;
+use PHPUnit\Framework\TestCase;
+
+// Creates a standard JSON auth object for testing.
+function createISACTestJson()
```

Comment:
Would you mind if we make this a private method on the test class below? It also would be nice to have a test in [ApplicationDefaultCredentialsTest](https://github.com/googleapis/google-auth-library-php/blob/main/tests/ApplicationDefaultCredentialsTest.php) covering the case of instantiation of an `ImpersonatedServiceAccountCredentials` instance. If you'd like us to help out with adding that, let us know :).